### ISSUE TITLE
Consider tagging a new version?

### ISSUE THREAD

**Issue by hosiet** [goldendict/goldendict#839]
It's been over a year since last released version/tag (1.5.0-rc2).

It would be great if upstream could make tags more often to ease downstream maintainance burden. Or at least I'm looking forward to a 1.5.0-rc3 / 1.5.0 tag soon.

**Comment by Eosphorus** [comment]
Or, as I would suggest, drop the whole version thing altogether and simply give the app a date stamp as to when it was built.

As long as an app, formerly known as a program, is being improved, debugged, enhanced and moreover used, there'll never be a 'final version' of it.

I call my builds something like "version 2017-05-02" - that would be today, today.

Just a thought.

**Comment by hosiet** [comment]
Using timestamp is totally fine. Yet making tags can be beneficial to downstream. E.g., fetching tarballs.

We could tag monthly snapshots in that case and call it a release. Anyway that helps downstream packagers.

**Comment by timofonic** [comment]
I agree. Using timestamps would be a great idea. I also would like more releases and then report found bugs instead slow releases.

**Comment by hosiet** [comment]
@Abs62 It has been almost 3 years but we are still stuck here: the lack of new release (or tags) of goldendict is seriously causing troubles to downstream Linux distributions, especially that the last release was 5 years ago.

Generating a tag with higher version number should be really easy and can be done in minutes. If for whatever reason you are not tagging anymore in the future, please do make such statement in the README file so that all downstream maintainers are aware of it and may shift to a snapshot-based workflow.

Doing this would be really easy but your attitude is extremely important to others users and package maintainers.

**Comment by vedgy** [comment]
Apparently no developer with requisite access rights is willing to tag new versions or release candidates. GoldenDict's *master* branch is generally very stable. Regressions are usually fixed promptly. How about "tagging" a community RC on a new dedicated Wiki page by associating the current latest commit in *master* with *1.5.0-RC2.1*?

**Comment by danfe** [comment]
> GoldenDict's _master_ branch is generally very stable. Regressions are usually fixed promptly.

This is awesome, but tagged releases (betas, candidates) help downstream maintainers to decide when to update the package.  Tracking every commit of the repo does not scale (we typically maintain _hundreds_ of different software with all sorts of upstream repositories and work styles), only developers would really know when is the right time to cut the release (e.g. after previous big change had matured, some annoying long-standing bug had been finally tracked down and fixed, or just before potentially disruptive change is about to hit the tree).

> How about "tagging" a community RC on a new dedicated Wiki page by associating the current latest commit in _master_ with _1.5.0-RC2.1_?

Well, no, this is not the tagging we (downstream packagers) are asking for, sorry.

**Comment by vedgy** [comment]
I recently became a member of GoldenDict team and plan to tag a new date-based version in a couple of weeks. @goldendict/developers, any objections?

**Comment by vedgy** [comment]
Actually I think we should just release the long-delayed version 1.5 for versioning simplicity. Then release patch versions, e.g. 1.5.1, for minor fixes and improvements; minor versions, e.g. 1.6, for larger changes. I am going to tag version 1.5 on Monday, May 29, unless someone objects or proposes a viable alternative.

@Abs62, would you be willing to attach fresh Windows builds to a GitHub release? If not, I suppose I can repack the 3 Windows archives myself based on instructions [here](https://forum.ru-board.com/topic.cgi?forum=5&topic=49149).

**Comment by hosiet** [comment]
@vedgy Please use `1.5.0` instead of `1.5`: `1.5.0 > 1.5`, and we already have `1.5.0-rc*` previously. Or just jump to anything `>= 1.5.1` to avoid all previous `1.5*` mess.

**Comment by vedgy** [comment]
OK, let it be 1.5.0.

**Comment by vedgy** [comment]
Just pushed the annotated tag `1.5.0`. A GitHub release should be created to improve the visibility and simplify downloading the new release binaries. I wrote above:
> @Abs62, would you be willing to attach fresh Windows builds to a GitHub release? If not, I suppose I can repack the 3 Windows archives myself based on instructions here.

However, after writing that comment I realized that the GoldenDict version shown in the *About* dialog would be wrong if I repack an executable published on May 19. @Abs62, could you please make Windows builds based on the tagged commit, create a GitHub release and attach the builds to it? Or just make and share the builds to let me create a GitHub release?

**Comment by Abs62** [comment]
>> Or just make and share the builds to let me create a GitHub release?

Builds uploaded to [sourceforge.net](https://sourceforge.net/projects/goldendict/files/goldendict/1.5/).

**Comment by vedgy** [comment]
> Builds uploaded to [sourceforge.net](https://sourceforge.net/projects/goldendict/files/goldendict/1.5/).

Thank you! I'll create a GitHub release tomorrow.

**Comment by vedgy** [comment]
Compared the archive contents with the latest early access builds:
1. The file *LICENSE.txt* is absent from *GoldenDict-1.5.0_(QT_5123)(64bit).7z*. Is this an unintentional omission?
2. The *opencc* directory and *mimetex.cgi* are absent from the installer version.

Should the release announcement say something like this?

----
**Note:** the installer contains the Qt4-based version of GoldenDict with two limitations:
1. No conversion between Traditional and Simplified Chinese.
2. No LaTeX formula support.
----
Another question: is the Qt4-based version 32-bit?

**Comment by Abs62** [comment]
>> The file LICENSE.txt is absent from GoldenDict-1.5.0_(QT_5123)(64bit).7z. Is this an unintentional omission?

No. Fixed.

>> The opencc directory and mimetex.cgi are absent from the installer version.

Too old script. Fixed.

>> is the Qt4-based version 32-bit?

Yes.

**Comment by vedgy** [comment]
Created the [GitHub release 1.5.0](https://github.com/goldendict/goldendict/releases/tag/1.5.0).
